cmd/aks-periscope/aks-periscope.go

```diff
@@ -40,45 +40,192 @@ func main() {
+}
+
+// selectCollectors select the collectors to run
+func selectCollectors(allCollectorsByName map[string]interfaces.Collector) []interfaces.Collector {
+	collectors := []interfaces.Collector{}
+
+	//read list of collectors that are enabled
+	var enabledCollectorNames []string
+
+	//TODO try get partners to move from COLLECTOR_LIST to use ENABLED_COLLECTORS instead, for now COLLECTOR_LIST takes precedence if defined
+	collectorList, found := os.LookupEnv("COLLECTOR_LIST")
+	if found {
+		enabledCollectorNames = selectCollectorsUsingCollectorList(strings.Fields(collectorList))
+	} else {
+		enabledCollectorNames = strings.Fields(os.Getenv("ENABLED_COLLECTORS"))
+	}
+
+	for _, collectorName := range enabledCollectorNames {
+		collectors = append(collectors, allCollectorsByName[collectorName])
 	}
 
-	if contains(collectorList, "connectedCluster") {
-		collectors = append(collectors, helmCollector)
+	return collectors
+}
+
+//selectCollectorsUsingCollectorList use clusterType
+func selectCollectorsUsingCollectorList(collectorList []string) []string {
+	var enabledCollectorNames []string
+
+	//select default collectors
+	enabledCollectorNames = append(enabledCollectorNames,
+		"dns", "containerlogs", "kubeobjects", "networkoutbound")
+
+	if utils.Contains(collectorList, "connectedCluster") {
+		//select connectedCluster collectors
+		enabledCollectorNames = append(enabledCollectorNames, "helm")
 	} else {
-		collectors = append(collectors, systemLogsCollector)
-		collectors = append(collectors, ipTablesCollector)
-		collectors = append(collectors, nodeLogsCollector)
-		collectors = append(collectors, kubeletCmdCollector)
-		collectors = append(collectors, systemPerfCollector)
+		//select non-connectedCluster collectors
+		enabledCollectorNames = append(enabledCollectorNames,
+			"iptables", "kubeletcmd", "nodelogs", "systemlogs", "systemperf")
+	}
+	if utils.Contains(collectorList, "OSM") {
+		//select OSM collectors
+		enabledCollectorNames = append(enabledCollectorNames, "osm")
 	}
 
-	if contains(collectorList, "OSM") {
-		collectors = append(collectors, osmCollector)
+	return enabledCollectorNames
+}
+
+// selectDiagnosers select the diagnosers to run
+func selectDiagnosers(allDiagnosersByName map[string]interfaces.Diagnoser) []interfaces.Diagnoser {
+	diagnosers := []interfaces.Diagnoser{}
+
+	//read list of diagnosers that are enabled
+	enabledDiagnoserString, found := os.LookupEnv("ENABLED_DIAGNOSERS")
+	if !found {
+		//if not defined, default to all diagnosers enabled
+		enabledDiagnoserString = "networkconfig networkoutbound"
 	}
 
-	collectorGrp := new(sync.WaitGroup)
+	enabledDiagnoserNames := strings.Fields(enabledDiagnoserString)
+
+	for _, diagnoserName := range enabledDiagnoserNames {
+		diagnosers = append(diagnosers, allDiagnosersByName[diagnoserName])
+	}
+
+	return diagnosers
+}
+
+// selectedExporters select the exporters to run
+func selectExporters(allExporters map[string]interfaces.Exporter) []interfaces.Exporter {
```

Comment:
This is out of the PR scope

Reply:
I understand but right now the PR is hard to review and you have to deal with our numerous comments.
I think exporters shall not be treated the same way as collectors and diagnosers: exporters shall be configured by the user (export to blob, to disk, ...) but as we already discussed, I believe collectors and diagnosers should be 'hidden' from the users.

May be before a PR to support Kind cluster we should address these points in a dedicated PR ?